Set-DomainUser.ps1

```diff
@@ -1,720 +1,1124 @@
+                backupSafe                             = $True
+                viewAuditLog                           = $True
+                viewSafeMembers                        = $True
+                accessWithoutConfirmation              = $True
+                createFolders                          = $True
+                deleteFolders                          = $True
+                moveAccountsAndFolders                 = $True
+                requestsAuthorizationLevel1            = $True 
+                requestsAuthorizationLevel2            = $False
+            }
+        }
+        $json = $body | ConvertTo-Json
+        $null = Invoke-RestMethod -Method 'Post' -Uri $url -Body $json -Headers @{ 'Authorization' = $pvwaToken } -ContentType 'application/json'
+    }
+    catch {
+        Write-Host $_.ErrorDetails.Message 
+    }
+}
+
+Function Check-UM {
+    <#
+    .SYNOPSIS
+    Checks to see if tenant is UM or not (from the connector server)
+    .DESCRIPTION
+    Checks to see if tenant is UM or not (from the connector server)
+    .PARAMETER psmRootInstallLocation
+    PSM Folder
+    #>
+    param (
+        [Parameter(Mandatory = $true)]
+        $psmRootInstallLocation
+    )
+    $psmBasicPSMContent = Get-Content -Path $psmRootInstallLocation\basic_psm.ini 
+    $validation = $psmBasicPSMContent -match "IdentityUM.*=.*Yes"
+    return ("" -ne $validation) 
+}
+
+#Running Set-DomainUser script
+
+if ($null -eq $psmConnectCredentials) {
+    $psmConnectCredentials = Get-Credential -Message "Please enter PSMConnect domain user credentials"
+    if (!($psmConnectCredentials)) {
+        Write-Error "No credentials provided. Exiting."
+        exit 1
+    }
+}
+
+if ($null -eq $psmAdminCredentials) {
+    $psmAdminCredentials = Get-Credential -Message "Please enter PSMAdminConnect domain user credentials"
+    if (!($psmAdminCredentials)) {
+        Write-Error "No credentials provided. Exiting."
+        exit 1
+    }
+}
+
+$REGKEY_PSMSERVICE = "Cyber-Ark Privileged Session Manager"
+$psmRootInstallLocation = ($(Get-ServiceInstallPath $REGKEY_PSMSERVICE)).Replace("CAPSM.exe", "").Replace('"', "").Trim()
+
+If (Check-UM -psmRootInstallLocation $psmRootInstallLocation) {
+    $UM = $true
+}
+else { 
+    $UM = $false
+}
+
+if ($null -eq $tinaCreds) {
+    if ($UM) {
+        $TinaUserType = "installer user"
+    }
+    else {
+        $TinaUserType = "tenant administrator"
+    }
+    $tinaCreds = Get-Credential -Message ("Please enter {0} credentials" -f $TinaUserType)
+    if (!($tinaCreds)) {
+        Write-Error "No credentials provided. Exiting."
+        exit 1
+    }
+}
+
+[Net.ServicePointManager]::SecurityProtocol = [Net.SecurityProtocolType]::Tls12
+
+$BackupSuffix = (Get-Date).ToString('yyyMMdd-HHmmss')
+$DomainNameAutodetected = $false
+
+$Tasks = @(
+    "Configure PSM to use the domain PSM accounts"
+    "Modify local/group policies to allow PSM users to use Remote Desktop"
+)
```

Comment:
We'll build and add onto an array of strings which will be displayed at the end to give the user a list of tasks to perform before the process can be considered complete